src/live/handleAlbumChat.js

```diff
@@ -1,11 +1,37 @@
-function handleAlbumChat(io, socket, data) {
-  const { message, userId, albumId } = data;
+const ChatMessage = require('../models/ChatMessage');
 
-  // Create a chat room name for the album chat
-  const chatRoomName = `chat:album:${albumId}`;
+async function handleAlbumChat(io, socket, data) {
+  try {
+    const { albumId } = data;
 
-  // Send the message to the room
-  socket.to(chatRoomName).emit('chat:album', { message, userId, albumId });
+    // Create a chat room name for the album chat
+    const chatRoomName = `chat:album:${albumId}`;
+
+    const msg = await ChatMessage.create({ ...data, username: data.user });
+    await msg.populate('user', 'username');
+    console.log('Chat message saved:', msg);
```

Comment:
Nice! You should see that when this is logged you get something like:

```javascript
Chat message saved: {
  albumId: '6r1lh7fHMB499vGKtIyJLy',
  message: 'hi',
  user: { _id: new ObjectId("64ef5e90086e1c19b888c72c"), username: 'akos' },
  _id: new ObjectId("6525be96a91bfb376e6e86d8"),
  timestamp: 2023-10-10T21:13:58.157Z,
  __v: 0
}
```

where `msg.user` is an object, that has a `msg.user.username` field that is the username of the person who sent this message :tada: